### ISSUE TITLE
execution reverted: Insufficient WETH9

### ISSUE THREAD

**Issue by LonelyNightCoder** [uniswap-python/uniswap-python#249]
**Describe the bug**
I have successfully exchanged ETH to USDS in Arbitrum Network, but i'm trying to implement an exchange of USDS to ETH too, but it raise error

**To Reproduce**
```
    ETH_ADDRESS = Web3.toChecksumAddress("0x88***")
    USDCToken      = Web3.toChecksumAddress("0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8")

    w3 = Web3(Web3.HTTPProvider('https://arb1.arbitrum.io/rpc', request_kwargs={"timeout": 60}))
    print(w3.isConnected())
    w3.eth.setGasPriceStrategy(slow_gas_price_strategy)

    Uniswap = uniswap.Uniswap(address=ETH_ADDRESS, 
                      private_key='KEY', 
                      version=3,
                      use_estimate_gas=True, web3=w3)

    qtyUSDC = int(0.4*10**6)
    impact = Uniswap.get_price_input(USDCToken, ETH_ADDRESS,  qtyUSDC, fee=500)
    print(f"Impact: {impact}")

    tx = Uniswap.make_trade(USDCToken, ETH_ADDRESS, qtyUSDC, fee=500)
    Uniswap.w3.eth.wait_for_transaction_receipt(tx, timeout=5)
```

**Expected behavior**
Successfully exchange USDC to ETH

**Error**
```
Traceback (most recent call last):
  File "main.py", line 63, in <module>
    main()
  File "main.py", line 51, in main
    tx = Uniswap.make_trade(USDCToken, ETH_ADDRESS, qtyUSDC, fee=500)
  File "/usr/local/lib/python3.8/dist-packages/uniswap/decorators.py", line 30, in approved
    return method(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/uniswap/uniswap.py", line 422, in make_trade
    return self._token_to_eth_swap_input(
  File "/usr/local/lib/python3.8/dist-packages/uniswap/uniswap.py", line 638, in _token_to_eth_swap_input
    return self._build_and_send_tx(
  File "/usr/local/lib/python3.8/dist-packages/uniswap/uniswap.py", line 1107, in _build_and_send_tx
    transaction = function.buildTransaction(tx_params)
  File "/usr/local/lib/python3.8/dist-packages/web3/contract.py", line 1079, in buildTransaction
    return build_transaction_for_function(
  File "/usr/local/lib/python3.8/dist-packages/web3/contract.py", line 1648, in build_transaction_for_function
    prepared_transaction = fill_transaction_defaults(web3, prepared_transaction)
  File "cytoolz/functoolz.pyx", line 250, in cytoolz.functoolz.curry.__call__
  File "/usr/local/lib/python3.8/dist-packages/web3/_utils/transactions.py", line 114, in fill_transaction_defaults
    default_val = default_getter(web3, transaction)
  File "/usr/local/lib/python3.8/dist-packages/web3/_utils/transactions.py", line 60, in <lambda>
    'gas': lambda web3, tx: web3.eth.estimate_gas(tx),
  File "/usr/local/lib/python3.8/dist-packages/web3/eth.py", line 855, in estimate_gas
    return self._estimate_gas(transaction, block_identifier)
  File "/usr/local/lib/python3.8/dist-packages/web3/module.py", line 57, in caller
    result = w3.manager.request_blocking(method_str,
  File "/usr/local/lib/python3.8/dist-packages/web3/manager.py", line 198, in request_blocking
    return self.formatted_response(response,
  File "/usr/local/lib/python3.8/dist-packages/web3/manager.py", line 170, in formatted_response
    apply_error_formatters(error_formatters, response)
  File "/usr/local/lib/python3.8/dist-packages/web3/manager.py", line 70, in apply_error_formatters
    formatted_resp = pipe(response, error_formatters)
  File "cytoolz/functoolz.pyx", line 667, in cytoolz.functoolz.pipe
  File "cytoolz/functoolz.pyx", line 642, in cytoolz.functoolz.c_pipe
  File "/usr/local/lib/python3.8/dist-packages/web3/_utils/method_formatters.py", line 576, in raise_solidity_error_on_revert
    raise ContractLogicError(response['error']['message'])
web3.exceptions.ContractLogicError: execution reverted: Insufficient WETH9
```

**Comment by LonelyNightCoder** [comment]
@ericjang Good afternoon, I need a little help, I still don’t understand the behavior of WETH9, when the withdrawal is on the ETH wallet

**Comment by GEEGABYTE1** [comment]
I am also having the same error when I interact with my own token on Ropsten.

**Comment by LonelyNightCoder** [comment]
> I am also having the same error when I interact with my own token on Ropsten.

I've already solved issue, i can help u, send message maximdevops@gmail.com

**Comment by ErikBjare** [comment]
@LonelyNightCoder Then say so here. Don't move the issue into a private email conversation.

**Comment by zaneran9** [comment]
I tried to execute my transaction using multicall(exactInputSingle, unwrapWETH9), and it returned execution reverted: "Insufficient WETH9. Obviously, the first step was successful, and the problem was with unwrapWETH9;

I tested using the WETH9 contract to execute the withdraw method, and it was able to be withdrawn, but I used the Router contract unwrapWETH9 method to execute it, and then it returned the above error (here is my transaction record https://sepolia.etherscan.io/tx/0xfd28c046a6e41a4ac0cbf1be7219a9bcf3c2c125ae9af2ac5129c9f8f36100b1), any suggestions? Thanks! @LonelyNightCoder
